File: backend/src/main/java/com/gpuoj/dto/JobMessage.java
package com.gpuoj.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class JobMessage {
    private String submissionId;
    private String problemId;
    private String sourceCode;        // base64-encoded
    private String testCasesPath;
    private String judgeType;
    private int timeLimitMs;
    private Integer vramLimitMb;
    private Double speedupThreshold;
}

File: backend/src/main/kotlin/com/gpuoj/dto/JobMessage.kt
package com.gpuoj.dto

data class JobMessage(
    val submissionId: String = "",
    val problemId: String = "",
    val sourceCode: String = "",  // base64-encoded
    val testCasesPath: String = "",
    val judgeType: String = "",
    val timeLimitMs: Int = 0,
    val vramLimitMb: Int? = null,
    val speedupThreshold: Double? = null
)
